### ISSUE TITLE
Have you tried the model on real world images?

### ISSUE THREAD

**Issue by arnonkahani** [akirasosa/mobile-semantic-segmentation#15]
Hi,
I have tried running produced from the training on sample images from normal real world seances, and I'm getting very bad results and idea why? the training and validation scores seem alright?

**Comment by akirasosa** [comment]
@arnonkahani 
> Have you tried the model on real world images?

Yes.

> I'm getting very bad results and idea why?

One of the possible reasons is just a coding miss in prediction. Have you already tried predicting photos which were used for training? At least, it must work good. Next, you will try to predict real photo without changing code, just replacing photo.

**Comment by arnonkahani** [comment]
What I actually meant was that I was getting bad results on images where the person was not in the middle of the picture. More like pictures of people in scenes where you could see there full body or multiple people.

**Comment by arnonkahani** [comment]
And another unrelated question, if I want to turn this for binary segmentation (hair/no hair) to face,hair and background how would I go around doing so?

Thanks

**Comment by akirasosa** [comment]
@arnonkahani 
> What I actually meant was that I was getting bad results on images where the person was not in the middle of the picture.

It's because there is no such kind of training data in LFW. As you can see on training data, person is always centered and we train machine to predict the region of hair of the person.

> And another unrelated question, if I want to turn this for binary segmentation (hair/no hair) to face,hair and background how would I go around doing so?

Then, you will have 3 classes. The output shape will be (h, w, 3). You apply softmax to calculate dice loss.
